application/app_config.py

```diff
@@ -4,80 +4,170 @@
     CURRENT_PROJECT_NAME = None
-    CURRENT_PROJECT_CONFIG = None  # Path
-    CURRENT_PROJECT_VIDEO_PATH = None
 
-    def __init__(self):
-        super(AppConfig, self).__init__()
+def get_base_project_dir():
+    return AppConfig.PROJECT_DIR
 
-    @classmethod
-    def load_application_config(cls):
-        config_parser = SafeConfigParser()
-        config_parser.read(os.path.join(os.path.dirname(os.path.realpath(__file__)), ".application"))
-        cls.PROJECT_DIR = config_parser.get("info", "default_project_dir")
+def get_project_path():
+    if AppConfig.CURRENT_PROJECT_NAME:
+        return os.path.join(AppConfig.PROJECT_DIR, AppConfig.CURRENT_PROJECT_NAME)
+    return None
 
-    # TODO: class method for writing to application config file
+def get_config_path():
+    if AppConfig.CURRENT_PROJECT_NAME:
+        return os.path.join(AppConfig.PROJECT_DIR, AppConfig.CURRENT_PROJECT_NAME, "config.cfg")
+    return None
 
-def update_project_cfg(section, option, value):
+def get_identifier():
+    config_path = get_config_path()
+    if config_path:
+        identifier_exists, identifier = get_config_with_sections(config_path, "info", "identifier")
+        return identifier
+    return None
+
+def get_project_video_path():
+    config_path = get_config_path()
+    if config_path:
+        (success, video) = get_config_with_sections(config_path, "video", "name")
+        if success:
+            return os.path.join(get_project_path(), video)
+        else:
+            print("ERR: project_video(): Couldn't get video")
+            return None
+    return None
+
+def update_config_with_sections(config_path, section, option, value):
     """
     Updates a single value in the current open project's configuration file.
     Writes nothing and returns -1 if no project currently open. Creates sections
     in the config file if they do not already exist.
 
     Args:
+        config_path (str): Path to the config file
         section (str): Name of the section to write new option-value pair to write.
         option (str): Name of the option to write/update.
         value (str): Value to write/update assocaited with the specified option.
     """
-    if not AppConfig.CURRENT_PROJECT_CONFIG:
+    if not os.path.exists(config_path):
+        print("ERR [update_config_with_sections()]: File {} does not exist.".format(config_path))
         return -1
     cfp = SafeConfigParser()
-    cfp.read(AppConfig.CURRENT_PROJECT_CONFIG)
+    cfp.read(config_path)
     if section not in cfp.sections():  # If the given section does not exist,
         cfp.add_section(section)        # then create it.
     cfp.set(section, option, value)  # Set the option-value pair
-    with open(AppConfig.CURRENT_PROJECT_CONFIG, "wb") as cfg_file:
+    with open(config_path, "wb") as cfg_file:
         cfp.write(cfg_file)  # Write changes
 
 
-def check_project_cfg_option(section, option):
+def get_config_with_sections(config_path, section, option):
```

Comment:
The function name of `get_config_with_sections` vs `get_config_section` does not help me remember the difference.  I'm having trouble coming up with a better name... but these are not good :(

`get_option_from_section_in_config(config_path, section, option)`

`get_entire_section_in_config(config_path, section)`

How do you like that?